example/tutorial_quanconv_mnist.py

```diff
@@ -0,0 +1,105 @@
+#! /usr/bin/python
+# -*- coding: utf-8 -*-
+
+import time
+import tensorflow as tf
+import tensorlayer as tl
+
+tf.logging.set_verbosity(tf.logging.DEBUG)
+tl.logging.set_verbosity(tl.logging.DEBUG)
+
+X_train, y_train, X_val, y_val, X_test, y_test = tl.files.load_mnist_dataset(shape=(-1, 28, 28, 1))
+# X_train, y_train, X_test, y_test = tl.files.load_cropped_svhn(include_extra=False)
+
+sess = tf.InteractiveSession()
+
+batch_size = 128
+
+x = tf.placeholder(tf.float32, shape=[batch_size, 28, 28, 1])
+y_ = tf.placeholder(tf.int64, shape=[batch_size])
+
+
+def model(x, is_train=True, reuse=False):
+    with tf.variable_scope("quan_cnn", reuse=reuse):
+        net = tl.layers.InputLayer(x, name='input')
+        net = tl.layers.QuanConv2dWithBN(
+            net, 32, (5, 5), (1, 1), padding='SAME', act=tf.nn.relu, is_train=is_train, name='qcbnb1'
+        )
+        net = tl.layers.MaxPool2d(net, (2, 2), (2, 2), padding='SAME', name='pool1')
+
+        net = tl.layers.QuanConv2dWithBN(
+            net, 64, (5, 5), (1, 1), padding='SAME', act=tf.nn.relu, is_train=is_train, name='qcbn2'
+        )
+        net = tl.layers.MaxPool2d(net, (2, 2), (2, 2), padding='SAME', name='pool2')
+
+        net = tl.layers.FlattenLayer(net)
+        # net = tl.layers.DropoutLayer(net, 0.8, True, is_train, name='drop1')
+        net = tl.layers.QuanDenseLayerWithBN(net, 256, is_train=is_train, act=tf.nn.relu, name='qdbn')
+
+        # net = tl.layers.DropoutLayer(net, 0.8, True, is_train, name='drop2')
+        net = tl.layers.QuanDenseLayer(net, 10, name='qdbn_out')
+    return net
+
+
+# define inferences
+net_train = model(x, is_train=True, reuse=False)
+net_test = model(x, is_train=False, reuse=True)
+
+# cost for training
+y = net_train.outputs
+cost = tl.cost.cross_entropy(y, y_, name='xentropy')
+
+# cost and accuracy for evalution
+y2 = net_test.outputs
+cost_test = tl.cost.cross_entropy(y2, y_, name='xentropy2')
+correct_prediction = tf.equal(tf.argmax(y2, 1), y_)
+acc = tf.reduce_mean(tf.cast(correct_prediction, tf.float32))
+
+# define the optimizer
+train_params = tl.layers.get_variables_with_name('quan_cnn', True, True)
+train_op = tf.train.AdamOptimizer(learning_rate=0.0001).minimize(cost, var_list=train_params)
+
+# initialize all variables in the session
+tl.layers.initialize_global_variables(sess)
```

Comment:
tl.layers.initialize_global_variables is deprecated in favor of `tf.global_variables_initializer`

Reply:
+1